genie-web/src/main/java/com/netflix/genie/web/configs/GenieTasksAutoConfiguration.java

```diff
@@ -0,0 +1,233 @@
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.configs;
+
+import com.netflix.genie.common.internal.util.GenieHostInfo;
+import com.netflix.genie.web.properties.ClusterCheckerProperties;
+import com.netflix.genie.web.properties.DatabaseCleanupProperties;
+import com.netflix.genie.web.properties.DiskCleanupProperties;
+import com.netflix.genie.web.properties.JobsProperties;
+import com.netflix.genie.web.properties.TasksExecutorPoolProperties;
+import com.netflix.genie.web.properties.TasksSchedulerPoolProperties;
+import com.netflix.genie.web.services.ClusterPersistenceService;
+import com.netflix.genie.web.services.FilePersistenceService;
+import com.netflix.genie.web.services.JobPersistenceService;
+import com.netflix.genie.web.services.JobSearchService;
+import com.netflix.genie.web.services.TagPersistenceService;
+import com.netflix.genie.web.tasks.leader.ClusterCheckerTask;
+import com.netflix.genie.web.tasks.leader.DatabaseCleanupTask;
+import com.netflix.genie.web.tasks.node.DiskCleanupTask;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.Executor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.actuate.autoconfigure.endpoint.web.WebEndpointProperties;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+import org.springframework.core.task.AsyncTaskExecutor;
+import org.springframework.core.task.SyncTaskExecutor;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.IOException;
+
+/**
+ * Configuration of beans for asynchronous tasks within Genie.
+ *
+ * @author tgianos
+ * @since 3.0.0
+ */
+@Configuration
+@EnableScheduling
+@EnableConfigurationProperties(
+    {
+        ClusterCheckerProperties.class,
+        DatabaseCleanupProperties.class,
+        DiskCleanupProperties.class,
+        TasksExecutorPoolProperties.class,
+        TasksSchedulerPoolProperties.class
+    }
+)
+public class GenieTasksAutoConfiguration {
```

Comment:
Its often help to add a `@ConditionalOnProperty` to autoconfigs like this. Somewhere down the line you may want to disable scheduled tasks for a particular stack. You can always get around it by setting the `spring.autoconfigure.exclude ` property. Mostly a style difference, but something to consider.

Reply:
There's still some interplay between the configs in so much as some beans in here are used in others. I'd like to detangle that web but probably not going to get to it right now